File: TDDMicroExercises.ProposedSolution/CSharp/TurnTicketDispenser/TurnNumberSequence.cs
namespace TDDMicroExercises.OneSolution.TurnTicketDispenser
{
    public class TurnNumberSequence : ITurnNumberSequence
    {
        public static readonly TurnNumberSequence GlobalTurnNumberSequence = new TurnNumberSequence();
        
        private int _turnNumber = 0;

        public int GetNextTurnNumber()
        {
            return _turnNumber++;
        }
    }
}

File: TDDMicroExercises.ProposedSolution/Java/TurnTicketDispenser/src/main/java/tddmicroexercises/turnticketdispenser/TurnNumberSequence.java
package tddmicroexercises.turnticketdispenser;

public class TurnNumberSequence implements ITurnNumberSequence
{
	public static ITurnNumberSequence turnNumberSequence = new TurnNumberSequence();

	private int _turnNumber = 0;

    public int getNextTurnNumber()
    {
        return _turnNumber++;
    }
}
